### ISSUE TITLE
Topic of project

### ISSUE THREAD

**Issue by Krzysztof-Cieslak** [devLDZ/FSharpSampleProject#1]
Let's discus.

One of the proposals was web-shop/e-commerce application.
Any other ideas?

**Comment by witchy-mutant** [comment]
Maybe session ratings (app used to gather videos from conferences together with ratings).

I like web-shop/e-commerce idea as well.

**Comment by Krzysztof-Cieslak** [comment]
E-commerce seems to have more interesting domain. We could use DDD / CQRS / ES ( especially since there will be some presentation about it in near future).

I just thought a bit about it but session ratings seems to be just simpler CRUD application.

**Comment by witchy-mutant** [comment]
So what would you prefer @Krzysztof-Cieslak?

**Comment by Krzysztof-Cieslak** [comment]
@mihcall, Well, I personally like DDD / CQRS so I would choose e-commerce. Also other applications with domain which contains something more than just simple create/update stuff.

Restaurant/Cafeteria/Gastronomy application can also be decent idea (and is one of classic CQRS examples).

Logistic/transport company/courier application can be OK.

Simple bank application seems good.

There are many good possible topics.

Still, I don't want to push my opinions on it very hard, since i have bit more experience with F# than other guys.

**Comment by theimowski** [comment]
I'll also share my idea for a project that once came to my mind: 
**GitHub repo code-quality rating system**

Main motivation for the idea is that when a developer is browsing web for best examples/practices/patterns of a particular language or technology, he will find plenty of resources more or less subjective. It's hard, however, to point to a specific codebase that could act as example of usage of the language/technology. 

Major use case for such an application could be then as following:
Find 10 top user-rated repositories on GitHub for language _LANG_.

I'm aware that one could search directly on GitHub with the `explore` dashboard, or search for repositiories with the highest amount of stars, but those are general-purpose ratings, and tend to mean that a repository is better known rather than of higher code quality.

What you think?

**Comment by iskeld** [comment]
Maybe a code review tool ?

**Comment by witchy-mutant** [comment]
cool idea @iskeld!

**Comment by senssei** [comment]
Or voting tool?

**Comment by witchy-mutant** [comment]
I like @theimowski idea but I don't see an objective criteria for judging projects. That's why sharing good examples of github repositiories is happening mostly on twitter with annonation that this is a subjective opinion of an individual :)

**Comment by witchy-mutant** [comment]
Voting tool is a good idea as well :)

**Comment by Krzysztof-Cieslak** [comment]
Those ideas are OK. 

Still I would put my money on e-commerce ;)

**Comment by theimowski** [comment]
@mihcall the rating doesn't have to objective. Stastically, a lot of subjective ratings could result in an overall objective rating - I am curious if it could work
I don't insist however on the idea, I like other proposals as well ;)

**Comment by Krzysztof-Cieslak** [comment]
So summing up, as for now we have following proposals:
1. Code review tool
2. e-commerce
3. GitHub repo code-quality rating system
4. Voting tool

Anything more?

**Comment by witchy-mutant** [comment]
@theimowski agreed ;)

**Comment by witchy-mutant** [comment]
so should we set up voting tool to vote on a next project? :)

**Comment by senssei** [comment]
:]

śr., 26 lis 2014, 17:40 Michał Śliwoń użytkownik notifications@github.com
napisał:

**Comment by Krzysztof-Cieslak** [comment]
Well, i'm repeating myself, but i think e-commerce application have most interesting domain, gives possibility of using CQRS/ES. While others are just relatively simple tools where most focus is on technical problem.

But it was idea of @kubawalinski and he haven't said anything yet here ;)

**Comment by fzawada** [comment]
How about
5. Kanban board
or a bigger beast
6. SonarQube-like system for .NET (code metrics, etc) 

E-commerce domain is I guess familiar to many but quite boring at the same time ;)

**Comment by Krzysztof-Cieslak** [comment]
Well, it really depends on what we want to achieve - if we want to create good f# demo application (and there is no good complex f# demo app in the web, those online focus just on some part as modeling, DAL etc) I would stick to topic maybe more boring but easy to understand for others.

If, to addition we practice f#, we want to create something cool, we can use our creativity and create something very interesting but harder to implement  ( like idea 6. )

**Comment by senssei** [comment]
Guys, we can start more than one project in parallel and contribute individually, even to more than one project.

**Comment by kubawalinski** [comment]
I really like the eCommerce idea but if we want to write a sample fsharp app then I think it's best not to mix in CQRS, EventSourcing or anything like that. It will unnecessarily complicate the problem. The language is new for most of us, so let's stick to the tried and tested paradigms for now.

**Comment by iskeld** [comment]
As @senssei said, I think it would be best if everyone who has an idea, some free time and will, started a project so that, interested contributors could join. So just start a project you like and let the community decide where to join :).

**Comment by kubawalinski** [comment]
@iskeld +1!

**Comment by Krzysztof-Cieslak** [comment]
OK, so someone has to start. I've pushed a project based on known CQRS Cafe sample from http://cqrs.nu/. It covers DDD, CQRS, ES, using EventStore as database. Focus is put on functional programming style - it's not just rewritten F# object oriented application ( what also would be possible). 

I've created some functional abstraction layer on top of EventStore and bases to make DDD style modeling easier.
